### ISSUE TITLE
Port twisted.application.service on py3 for trial

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#7816]
|[<img alt="adiroiban's avatar" src="https://avatars.githubusercontent.com/u/204609?s=50" width="50" height="50">](https://github.com/adiroiban)| @adiroiban reported|
|-|-|
|Trac ID|trac#7816|
|Type|enhancement|
|Created|2015-03-16 11:09:40Z|
|Branch|https://github.com/twisted/twisted/tree/py3-application-service-7816|

Duplicate of [#6916](https://github.com/twisted/twisted/issues/6916)

<details><summary>Searchable metadata</summary>

```
trac-id__7816 7816
type__enhancement enhancement
reporter__adiroiban adiroiban
priority__normal normal
milestone__Python_3_x Python-3.x
branch__branches_py3_application_service_7816 branches/py3-application-service-7816
branch_author__ 
status__closed closed
resolution__duplicate duplicate
component__core core
keywords__ 
time__1426504180244124 1426504180244124
changetime__1435821276898382 1435821276898382
version__None None
owner__mesozoic mesozoic

```
</details>

**Comment by twisted-trac** [comment]
|[<img alt="mesozoic's avatar" src="https://avatars.githubusercontent.com/u/458347?s=50" width="50" height="50">](https://github.com/mesozoic)<a name="note_1"></a>|@mesozoic set owner to @mesozoic<br>@mesozoic set status to `assigned`|
|-|-|

I thought I could get through this tonight, but I've gotta walk away and get some rest. Getting `twisted.test.test_application` to pass under Python 3 involved more mucking around than I thought it would. The branch is up-to-date. 

I'm a bit stuck on this test failure:

```
$ python admin/run-python3-tests twisted.test.test_application.LoadingTests.test_simpleStoreAndLoad
test_simpleStoreAndLoad (twisted.test.test_application.LoadingTests)
test_simpleStoreAndLoad ... ERROR

======================================================================
ERROR: test_simpleStoreAndLoad (twisted.test.test_application.LoadingTests)
test_simpleStoreAndLoad
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/Users/mesozoic/Development/ext/twisted/twisted/twisted/internet/defer.py", line 150, in maybeDeferred
    result = f(*args, **kw)
  File "/Users/mesozoic/Development/ext/twisted/twisted/twisted/internet/utils.py", line 201, in runWithWarningsSuppressed
    reraise(exc_info[1], exc_info[2])
  File "/Users/mesozoic/Development/ext/twisted/twisted/twisted/python/compat.py", line 374, in reraise
    raise exception.with_traceback(traceback)
  File "/Users/mesozoic/Development/ext/twisted/twisted/twisted/internet/utils.py", line 197, in runWithWarningsSuppressed
    result = f(*a, **kw)
  File "/Users/mesozoic/Development/ext/twisted/twisted/twisted/test/test_application.py", line 232, in test_simpleStoreAndLoad
    p.save()
  File "/Users/mesozoic/Development/ext/twisted/twisted/twisted/persisted/sob.py", line 124, in save
    self._saveTemp(filename, passphrase, dumpFunc)
  File "/Users/mesozoic/Development/ext/twisted/twisted/twisted/persisted/sob.py", line 95, in _saveTemp
    dumpFunc(self.original, f)
  File "/Users/mesozoic/Development/ext/twisted/twisted/twisted/persisted/aot.py", line 463, in jellyToSource
    aot = jellyToAOT(obj)
  File "/Users/mesozoic/Development/ext/twisted/twisted/twisted/persisted/aot.py", line 454, in jellyToAOT
    return AOTJellier().jelly(obj)
  File "/Users/mesozoic/Development/ext/twisted/twisted/twisted/persisted/aot.py", line 566, in jelly
    ao = self.jellyToAO(obj)
  File "/Users/mesozoic/Development/ext/twisted/twisted/twisted/persisted/aot.py", line 559, in jellyToAO
    raise TypeError("Unsupported type: %s" % objType.__name__)
TypeError: Unsupported type: Componentized

----------------------------------------------------------------------
Ran 1 test in 0.007s

FAILED (errors=1)
```

...and can't tell if that's because `twisted.persisted` needs more work, or if it's a regression I've introduced somewhere, or what. But I'll take another look later this week.

**Comment by twisted-trac** [comment]
|[<img alt="mesozoic's avatar" src="https://avatars.githubusercontent.com/u/458347?s=50" width="50" height="50">](https://github.com/mesozoic)<a name="note_3"></a>|@mesozoic commented|
|-|-|

This ticket is waiting on [#7827](https://github.com/twisted/twisted/issues/7827) (porting twisted.persisted.styles)

**Comment by twisted-trac** [comment]
|[<img alt="hawkowl's avatar" src="https://avatars.githubusercontent.com/u/3307100?s=50" width="50" height="50">](https://github.com/hawkowl)|@hawkowl set status to `closed`|
|-|-|
